File: app/src/main/java/com/example/mobv/Adapter/RoomAdapter.java
package com.example.mobv.Adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.mobv.MessageActivity;
import com.example.mobv.Model.ChatRoom;
import com.example.mobv.R;
import com.example.mobv.api.responses.Contact;

import java.util.List;

public class RoomAdapter extends RecyclerView.Adapter<RoomAdapter.ViewHolder> {

    private Context mContext;
    private List<ChatRoom> chatRooms;

    public RoomAdapter(Context mContext, List<ChatRoom> chatRooms) {
        this.chatRooms = chatRooms;
        this.mContext = mContext;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.room_item, parent, false);

        return new RoomAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        final ChatRoom room = chatRooms.get(position);
        holder.roomName.setText(room.getName());

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(mContext, MessageActivity.class);
                intent.putExtra("roomid", room.getName());
                mContext.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return chatRooms.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{

        public TextView roomName;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            roomName = itemView.findViewById(R.id.roomName);
        }
    }

}

File: app/src/main/java/com/example/mobv/adapter/RoomAdapter.kt
package com.example.mobv.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.mobv.model.ChatRoom
import com.example.mobv.R
import com.example.mobv.RoomMessageActivity

class RoomAdapter(
    private val mContext: Context,
    private val chatRooms: List<ChatRoom>

) : RecyclerView.Adapter<RoomAdapter.ViewHolder>() {

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view =
            LayoutInflater.from(mContext).inflate(R.layout.room_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        val room = chatRooms[position]
        holder.roomName.text = room.getName()

        holder.itemView.setOnClickListener {
            val intent = Intent(mContext, RoomMessageActivity::class.java)
            intent.putExtra("room", room)
            mContext.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return chatRooms.size
    }

    inner class ViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        var roomName: TextView = itemView.findViewById(R.id.roomName)

    }

}